sqlmesh/core/context.py

```diff
@@ -855,22 +856,39 @@ def evaluate(
 
         return df
 
+    @overload
+    def format(
+        self,
+        *,
+        check: Literal[True],
+        **kwargs: t.Any,
+    ) -> bool: ...
+
+    @overload
+    def format(
```

Comment:
Why do we need these overloaded methods?

Reply:
For typing purposes. When `True` is passed as the `check` argument, the return value is a boolean indicating whether everything is correctly formatted or not. When `None` or `False` are passed, the return value is `None` (i.e. the same as it is currently). This preserves the current "API" of this method when not performing a check. If you prefer, I could just make the return value a boolean always.

Reply:
Ah, got it, makes sense. Didn't notice the return value.

FWIW, I think it's fine to always return a bool.

Reply:
fixed